bundles/org.openhab.binding.airgradient/src/main/java/org/openhab/binding/airgradient/internal/handler/AirGradientLocationHandler.java

```diff
@@ -0,0 +1,186 @@
+            Bridge bridge = getBridge();
+            if (bridge != null) {
+                AirGradientAPIHandler handler = (AirGradientAPIHandler) bridge.getHandler();
+                if (handler != null) {
+                    handler.pollingCode();
+                }
+            }
+        } else if (CHANNEL_LEDS_MODE.equals(channelUID.getId())) {
+            if (command instanceof StringType stringCommand) {
+                Bridge bridge = getBridge();
+                if (bridge != null) {
+                    AirGradientAPIHandler handler = (AirGradientAPIHandler) bridge.getHandler();
+                    if (handler != null) {
+                        handler.setLedMode(getSerialNo(), stringCommand.toFullString());
+                    }
+                }
+            } else {
+                logger.debug("Received command {} for channel {}, but it needs a string command", command.toString(),
+                        channelUID.getId());
+            }
+        } else if (CHANNEL_CALIBRATION.equals(channelUID.getId())) {
+            if (command instanceof StringType stringCommand) {
+                if ("co2".equals(stringCommand.toFullString())) {
+                    Bridge bridge = getBridge();
+                    if (bridge != null) {
+                        AirGradientAPIHandler handler = (AirGradientAPIHandler) bridge.getHandler();
+                        if (handler != null) {
+                            handler.calibrateCo2(getSerialNo());
+                        }
+                    }
+                } else {
+                    logger.debug(
+                            "Received unknown command {} for calibration on channel {}, which we don't know how to handle",
+                            command.toString(), channelUID.getId());
+                }
+            }
+        } else {
+            // This is read only
+            logger.debug("Received command {} for channel {}, which we don't know how to handle", command.toString(),
+                    channelUID.getId());
+        }
+    }
+
+    @Override
+    public void initialize() {
+        // set the thing status to UNKNOWN temporarily and let the background task decide for the real status.
+        // the framework is then able to reuse the resources from the thing handler initialization.
+        // we set this upfront to reliably check status updates in unit tests.
+        updateStatus(ThingStatus.UNKNOWN);
+
+        Bridge controller = getBridge();
+        if (controller == null) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.BRIDGE_UNINITIALIZED);
+        } else if (ThingStatus.OFFLINE.equals(controller.getStatus())) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.BRIDGE_OFFLINE);
+        } else {
+            updateStatus(ThingStatus.ONLINE);
+        }
+    }
+
+    public String getLocationId() {
+        AirGradientLocationConfiguration config = getConfigAs(AirGradientLocationConfiguration.class);
```

Comment:
I would suggest to get the config during init, check if it is valid and store it as a class level property.
Then this method could get removed and callers can directly access `config.location`

Reply:
Changed to cache config in both handlers.